datalad/tests/test_cmd.py

```diff
@@ -285,6 +285,18 @@ def test_runner_failure_unicode(path):
         runner.run(u"β-command-doesnt-exist", cwd=path)
 
 
+@skip_if_on_windows  # likely would fail
+@with_tempfile(mkdir=True)
+def test_runner_fix_PWD(path):
```

Comment:
This test passes for when I apply it, without the cmd.py changes, on top of this PR's base.

```
% git rev-parse HEAD
5d7014599a1476355df986d167f4b14689c1ff0c

% python -m nose -vs datalad/tests/test_cmd.py:test_runner_fix_PWD
datalad.tests.test_cmd.test_runner_fix_PWD ... ok
Versions: appdirs=1.4.3 boto=2.49.0 cmd:annex=7.20190219 cmd:git=2.21.0 cmd:system-git=2.21.0 cmd:system-ssh=7.4p1 exifread=2.1.2 git=2.1.11 gitdb=2.0.5 humanize=0.5.1 iso8601=0.1.12 keyring=18.0.0 keyrings.alt=3.1.1 msgpack=0.6.1 mutagen=1.42.0 requests=2.21.0 six=1.12.0 wrapt=1.11.1
Obscure filename: str=b' "\';a&b&c\xce\x94\xd0\x99\xd7\xa7\xd9\x85\xe0\xb9\x97\xe3\x81\x82 `| ' repr=' "\';a&b&cΔЙקم๗あ `| '
Encodings: default='utf-8' filesystem='utf-8' locale.prefered='UTF-8'
Environment: LANG='en_US.UTF-8' PYTHONPATH='/home/kyle/.local/lib/python' GIT_SSL_CAINFO='/home/kyle/.guix-profile/etc/ssl/certs/ca-certificates.crt' GIT_EXEC_PATH='/home/kyle/.guix-profile/libexec/git-core' PATH='/home/kyle/src/python/venvs/datalad/bin:/home/kyle/src/python/venvs/datalad//bin:/home/kyle/.guix-profile/bin:/home/kyle/.config/guix/current/bin:/home/kyle/.cabal/bin:/usr/lib/ccache/bin:/home/kyle/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games'

----------------------------------------------------------------------
Ran 1 test in 0.091s

OK
```

Reply:
strange -- will check

Reply:
ha -- it is (99% confidence guess) because test uses "shell=True" so sure thing that shell fixes it up.  So the fixup is needed only if shell is not True. Thanks for catching this!!!